### PR TITLE
default value of service account for DirectRunner and fix NPE for dat…

### PR DESCRIPTION
**Pull request by Nuzhdina-Elena**
Set default value for service account if runner is DirectRunner.
Check DataSchemaGcsPath is not null added.

### CODE REVIEW

File: v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java [hunk 1]
```diff
@@ -219,7 +226,9 @@ public static PipelineResult run(ProtegrityDataTokenizationOptions options,
       DataflowPipelineOptions dataflowOptions) {
     SchemasUtils schema = null;
     try {
-      schema = new SchemasUtils(options.getDataSchemaGcsPath(), StandardCharsets.UTF_8);
+      if (options.getDataSchemaGcsPath() != null) {
+        schema = new SchemasUtils(options.getDataSchemaGcsPath(), StandardCharsets.UTF_8);
+      }
```
Comment (KhaninArtur) [inline v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java]: Why did you add this null checking? Is it connected with some appeared warnings?

File: v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java [hunk 2]
```diff
@@ -206,6 +208,11 @@ public static void main(String[] args) {
         .withValidation()
         .as(DataflowPipelineOptions.class);
 
+    if (DirectRunner.class.isAssignableFrom(options.getRunner())) {
+      String serviceAccount = nullToEmpty(dataflowOptions.getServiceAccount());
+      dataflowOptions.setServiceAccount(serviceAccount);
+    }
```
Comment (KhaninArtur) [inline v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java]: I'm thinking - do we really need to check if it is a Direct runner specified? Probably, we can do something like this for simplicity:
```java
if (dataflowOptions.getServiceAccount() == null) {
  dataflowOptions.setServiceAccount("");
}
```

**Review by KhaninArtur — CHANGES_REQUESTED** [review]

File: v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java (hunk 2 above)
Comment (MikhailMedvedevAkvelon) [reply v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java:214]: We need have workaround about "org.apache.beam.runners.dataflow.options.DataflowPipelineOptions#getProject" option, that required for Dataflow runner, but not have sense for DirectRunner.

File: v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java [hunk 3]
```diff
@@ -206,6 +208,11 @@ public static void main(String[] args) {
         .withValidation()
         .as(DataflowPipelineOptions.class);
```
Comment (MikhailMedvedevAkvelon) [inline v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java]: We need have workaround about `org.apache.beam.runners.dataflow.options.DataflowPipelineOptions#getProject` option, that required for Dataflow runner, but not have sense for DirectRunner.

File: v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java (hunk 1 above)
Comment (MikhailMedvedevAkvelon) [reply v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java]: If `options.getDataSchemaGcsPath()` will be null, what we want to do? We want to LOG about it, we want to stop execution, we want to have some default value in that case?

**Review by MikhailMedvedevAkvelon — CHANGES_REQUESTED** [review]

File: v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java (hunk 1 above)
Comment (Nuzhdina-Elena) [reply v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java]: Ok, now if this property is null, throws NPE in SchemasUtils's constructor because there are transforms with this parameter.
So I suppose we should check it before and show correct message for user.

File: v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java (hunk 2 above)
Comment (Nuzhdina-Elena) [reply v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java]: Ok, but I'm afraid we will lose checking the service account for DataflowRunner when this parameter mustn't be null.

File: v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java (hunk 3 above)
Comment (Nuzhdina-Elena) [reply v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java]: Thank you for the notice. I'll fix it.

File: v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java (hunk 1 above)
Comment (KhaninArtur) [reply v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java]: Actually, we have a catch and check a bit lower that are supposed to validate the schema presence. But you've got a point, let's refactor this part like this:
```java
checkArgument(options.getDataSchemaGcsPath() != null, "Data schema is mandatory.");
SchemasUtils schema = null;
try {
  schema = new SchemasUtils(options.getDataSchemaGcsPath(), StandardCharsets.UTF_8);
}...
```

File: v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java (hunk 2 above)
Comment (KhaninArtur) [reply v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java]: You're right. The thing is that Dataflow takes care of Service Account itself, and we only need it to form a proper request to DSG. That's why we can simplify this 😁

File: v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java (hunk 3 above)
Comment (KhaninArtur) [reply v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java]: @MikhailMedvedevAkvelon does `getProject` throw an error in case if the project isn't specified?

File: v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java (hunk 1 above)
Comment (Nuzhdina-Elena) [reply v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java]: Thanks, fixed. Now use checkArgument.

File: v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java [hunk 4]
```diff
@@ -216,7 +224,9 @@ public static void main(String[] args) {
    * @return The pipeline result.
    */
   public static PipelineResult run(ProtegrityDataTokenizationOptions options,
-      DataflowPipelineOptions dataflowOptions) {
+      String serviceAccount) {
```
Comment (KhaninArtur) [inline v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java:228]: Could you please add this parameter to the docstring?

**Review by KhaninArtur — CHANGES_REQUESTED** [review]

File: v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java (hunk 3 above)
Comment (Nuzhdina-Elena) [reply v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java]: Rewrited this piece of code. Can you see, please? 
If it's DataflowRunner, we use DataflowPipelineOptions, whereas we deal with DirectRunner we don't need it.

File: v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java (hunk 4 above)
Comment (Nuzhdina-Elena) [reply v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/templates/ProtegrityDataTokenization.java:228]: Yes, of course. Done.

**Review by KhaninArtur — APPROVED** [review]

**Review by MikhailMedvedevAkvelon — APPROVED** [review]
